test/dotnet-performance.Tests/PerformanceTestBase.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Microsoft.DotNet.TestFramework;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Tools.Builder.Tests
+{
+    public class PerformanceTestBase : TestBase
+    {
+        protected void Build(string project)
+        {
+            Run(new BuildCommand(project, buildProfile: false));
+        }
+
+        protected void Run(TestCommand command)
+        {
+            command.Execute().Should().Pass();
+        }
+
+        protected void RemoveBin(string project)
+        {
+            Directory.Delete(Path.Combine(project, "bin"), true);
+        }
+
+        protected TestInstance CreateTestInstance(string testProjectName, string variation = "", [CallerMemberName] string callingMethod = "")
+        {
+            return TestAssetsManager.CreateTestInstance(Path.Combine("PerformanceTestProjects", testProjectName), callingMethod + variation)
+                .WithLockFiles();
+        }
+
+        public void Iterate(Action<PerformanceIterationContext> action, string variation = "", int iterations = 3, [CallerMemberName] string callingMethod = "")
```

Comment:
The test cases here are more comprehensive than the end-to-end test. We should converge on the BenchmarkAttribute infrastructure however. That one is already fairly developed and has broad ongoing investment.